mirrord/cli/src/connection.rs

```diff
@@ -120,40 +121,44 @@ where
         return Err(CliError::FeatureRequiresOperatorError("copy_target".into()));
     }
 
-    if matches!(
-        config.target,
-        mirrord_config::target::TargetConfig {
-            path: Some(
-                mirrord_config::target::Target::Deployment { .. }
-                    | mirrord_config::target::Target::Rollout(..)
-            ),
-            ..
-        }
+    match (
+        // user in mutipod without operator
+        matches!(
+            config.target,
+            mirrord_config::target::TargetConfig {
+                path: Some(
+                    mirrord_config::target::Target::Deployment { .. }
+                        | mirrord_config::target::Target::Rollout(..)
+                ),
+                ..
+            }
+        ),
+        // user using http filter(s) without operator
+        config.feature.network.incoming.http_filter.is_filter_set(),
     ) {
-        // Send to IDEs that we're in multi-pod without operator.
-        progress.ide(serde_json::to_value(IdeMessage {
-            id: MULTIPOD_WARNING.0.to_string(),
-            level: NotificationLevel::Warning,
-            text: MULTIPOD_WARNING.1.to_string(),
-            actions: {
-                let mut actions = HashSet::new();
-                actions.insert(IdeAction::Link {
-                    label: "Get started (read the docs)".to_string(),
-                    link: "https://mirrord.dev/docs/overview/teams/?utm_source=multipodwarn&utm_medium=plugin".to_string(),
-                });
-                actions.insert(IdeAction::Link {
-                    label: "Try it now".to_string(),
-                    link: "https://app.metalbear.co/".to_string(),
-                });
-
-                actions
-            },
-        })?);
-        // This is CLI Only because the extensions also implement this check with better messaging.
-        progress.print("When targeting multi-pod deployments, mirrord impersonates the first pod in the deployment.");
-        progress.print("Support for multi-pod impersonation requires the mirrord operator, which is part of mirrord for Teams.");
-        progress.print("You can get started with mirrord for Teams at this link: https://mirrord.dev/docs/overview/teams/?utm_source=multipodwarn&utm_medium=cli");
-    }
+        (true, true) => {
+            // only show user one of the two msgs - each user should always be shown same msg
+            if mid::get("mirrord")
```

Comment:
Maybe wrap this logic to a function just so it will understandable that this is for not spamming user with random messages (`user_persistent_random_message_select() -> bool`?)